src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -771,6 +777,46 @@ DukValue ScriptEngine::PositionToDuk(const CoordsXYZ& position)
     return DukValue::take_from_stack(ctx);
 }
 
+constexpr static const char* ExpenditureTypes[] = {
+    "ride_construction",
+    "ride_runningcosts",
+    "land_purchase",
+    "landscaping",
+    "park_entrance_tickets",
+    "park_ride_tickets",
+    "shop_sales",
+    "shop_stock",
+    "food_drink_sales",
+    "food_drink_stock",
+    "wages",
+    "marketing",
+    "research",
+    "interest",
+};
+
+std::string_view ScriptEngine::ExpenditureTypeToString(ExpenditureType expenditureType)
+{
+    auto index = static_cast<size_t>(expenditureType);
+    if (index < std::size(ExpenditureTypes))
+    {
+        return ExpenditureTypes[index];
+    }
+    return {};
+}
+
+ExpenditureType ScriptEngine::StringToExpenditureType(const std::string_view expenditureType)
+{
+    size_t i = 0;
+    for (auto sz : ExpenditureTypes)
```

Comment:
I think you could do 
```
auto it = std::find(ExpenditureTypes.begin(), ExpenditureTypes.end(), sz); 
if (it != ExpenditureTypes.end())
{
    return  static_cast<ExpenditureType>(std::distance(ExpenditureTypes.begin(),it));
}

```

Reply:
Just remember that though this looks cleaner, it iterates on the whole container twice, as both `find` and `distance` have ~`O(n)` complexity on worst case

Reply:
It doesn't iterate the list on `std::distance` if it is a random access iterator that std::array or std::vector would offer. Also distance wouldn't be doing string compares so is considerably less expensive.